daffodil-core/src/main/scala/org/apache/daffodil/core/compiler/Compiler.scala

```diff
@@ -110,7 +110,10 @@ final class ProcessorFactory private (
   override def onPath(xpath: String): DFDL.DataProcessor = sset.onPath(xpath)
 
   override def forLanguage(language: String): DFDL.CodeGenerator = {
-    Assert.usage(!isError)
+    Assert.argCheck(
```

Comment:
Does this want to be something other than "argCheck"? To me argCheck implies we are making sure the arg passed in is valid. For example, you might do something like `argCheck(language != null)`. Feels like we still want this to be `usage()` but a public API usage variant.

Reply:
Thoughts on precondition()?

Reply:
What if we are just very direct with the name? This is a function that throws an IllegalStateException, so what about `illegalState()`? This way it can be used for either public or private functions where a failed test just wants to throw this one specific exception?